src/lib.rs

```diff
@@ -215,12 +215,13 @@ mod contract {
     /// Must match CHAIN_ID to make sure it's signed for given chain vs replayed from another chain.
     #[no_mangle]
     pub extern "C" fn submit() {
-        use crate::transaction::EthSignedTransaction;
-        use rlp::{Decodable, Rlp};
+        use crate::prelude::TryFrom;
+        use crate::transaction::EthTransaction;
 
         let input = sdk::read_input();
-        let signed_transaction =
-            EthSignedTransaction::decode(&Rlp::new(&input)).sdk_expect("ERR_INVALID_TX");
+
+        let EthTransaction::Legacy(signed_transaction) =
+            EthTransaction::try_from(input.as_slice()).sdk_expect("ERR_INVALID_TX");
```

Comment:
I guess as you already added custom parsing errors for TXs, we might use `sdk_unwrap()` which will specify the exact error why the provided TX is invalid. Though this will require implementing `AsRef<[u8]>` trait.

Reply:
Done in https://github.com/aurora-is-near/aurora-engine/pull/165/commits/0d67c67be44d6b8fddc7b81bc88cb9697564ae9c